runtime/vm/resolvesupport.cpp

```diff
@@ -398,7 +405,11 @@ resolveClassRef(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA cpIndex, UDAT
 
 	accessModifiers = resolvedClass->romClass->modifiers;
 	if (J9_ARE_ANY_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_INSTANTIABLE)) {
-		if (!J9ROMCLASS_ALLOCATES_VIA_NEW(resolvedClass->romClass)) {
+		if (!J9ROMCLASS_ALLOCATES_VIA_NEW(resolvedClass->romClass)
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+                || (J9_IS_J9CLASS_VALUETYPE(resolvedClass) && J9_ARE_NO_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_VALUETYPE))
```

Comment:
I think there is still an issue with this. 

The test write now says:
`if (notInstantiableViaNew || (isValueType && !resolveValueType)`

The problem is a valueType is not instantiable via new so the first part of the condition is always true